packages/neuron-wallet/src/services/indexer/transaction-cache.ts

```diff
@@ -0,0 +1,36 @@
+import { TransactionWithStatus } from "types/cell-types"
+
+export class TransactionCache {
```

Comment:
This `TransactionCache` looks like a Map has specified size`.

If so, you can use a proxy to trap the set method of a Map as follows

```javascript
const map = new Map()
map.set = new Proxy(map.set, {
  apply(target, thisArg, args) {
    if (target.name === 'set') {
      //pretreat
      console.log(args)
    }
    target.apply(thisArg, args)
  }
})
```

Reply:
Worth considering